lit_nlp/examples/models/dalle.py

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+
+import jax
+import  jax.numpy as jnp
+from functools import partial
+import random
+# Load models & tokenizer
+import dalle_mini 
+import vqgan_jax
+import vqgan_jax.modeling_flax_vqgan
+import transformers
+import flax
+import flax.training.common_utils
+import tqdm.notebook
+
+
+DalleBart = dalle_mini.DalleBart
+DalleBartProcessor = dalle_mini.DalleBartProcessor
+VQModel = vqgan_jax.modeling_flax_vqgan.VQModel
+CLIPProcessor = transformers.CLIPProcessor
+FlaxCLIPModel = transformers.FlaxCLIPModel
+shard_prng_key = flax.training.common_utils.shard_prng_key
+trange = tqdm.notebook.trange
+
+replicate = flax.jax_utils.replicate
+JsonDict = lit_types.JsonDict
+
+class DalleModel(lit_model.Model):
+  """Image to Text Model"""
+
+ 
+
+  def __init__(self,
+              model_name:Str,
+              predictions:int):
+    super().__init__()
+
+    self.model = model_name
+    self.predictions = predictions
+    
+  ##
+  # LIT API implementation
+  def max_minibatch_size(self) -> int:
+    return 8
+
+  def predict_minibatch(self, inputs):
+    """Model prediction based on code pipeline in doc https://github.com/borisdayma/dalle-mini"""
+    # tokenize the text. -> then return prediction
+    # Load VQGAN
+    # VQGAN model
+    VQGAN_REPO = "dalle-mini/vqgan_imagenet_f16_16384"
+    VQGAN_COMMIT_ID = "e93a26e7707683d349bf5d5c41c5b0ef69b677a9"
+
+    # small model -> dalle-mini/dalle-mini/mini-1:v0  larger one:"dalle-mini/dalle-mini/mega-1-fp16:latest"
+    DALLE_MODEL = self.model
+    DALLE_COMMIT_ID = None
+    model, params = DalleBart.from_pretrained(
+        DALLE_MODEL, revision=DALLE_COMMIT_ID, dtype=jnp.float16, _do_init=False
+    )
+    vqgan, vqgan_params = VQModel.from_pretrained(
+        VQGAN_REPO, revision=VQGAN_COMMIT_ID, _do_init=False
+    )
+    
+    params = replicate(params)
+    vqgan_params = replicate(vqgan_params)
+
+    # model inference
+    @partial(jax.pmap, axis_name="batch", static_broadcasted_argnums=(3, 4, 5, 6))
+    def p_generate(
+        tokenized_prompt, key, params, top_k, top_p, temperature, condition_scale
+    ):
+        return model.generate(
+            **tokenized_prompt,
+            prng_key=key,
+            params=params,
+            top_k=top_k,
+            top_p=top_p,
+            temperature=temperature,
+            condition_scale=condition_scale,
+        )
+
+    # decode image
+    @partial(jax.pmap, axis_name="batch")
+    def p_decode(indices, params):
+        return vqgan.decode_code(indices, params=params)
+    
+    # create a random key
+    seed = random.randint(0, 2**32 - 1)
+    key = jax.random.PRNGKey(seed)
+    
+    prompts = [ex["prompt"] for ex in inputs]
+
+    processor = DalleBartProcessor.from_pretrained(DALLE_MODEL, revision=DALLE_COMMIT_ID)
+    tokenized_prompts = processor(prompts)
+    tokenized_prompt = replicate(tokenized_prompts)
+
+    # generate Images
+    # number of predictions per prompt
+    n_predictions = self.predictions
```

Comment:
Rename `self.predictions` as `self.n_predictions` or similar as the former has a certain connotation in LIT (i.e., the literal predictions output from the model) and we don't want to overload that namespace.

Reply:
Also, if we want to suppor omre than 1 predicted image, we'll need to update a few other things:

- [ ] Add a `ImageBytesList` type to types.py (but need to wait for Crystal's type refactor PRs before adding this in this PR)
- [ ] Update [`GeneratedImageModule.supportedTypes`](https://github.com/PAIR-code/lit/blob/dev/lit_nlp/client/modules/generated_image_module.ts#L44) to include the `ImageBytesList`
- [ ] Update [`GeneratedImageModule.renderImage(...)`](https://github.com/PAIR-code/lit/blob/dev/lit_nlp/client/modules/generated_image_module.ts#L88) to accept `src: string | string[]` and use `Array.isArray(...)` to determine if we should use `src.map(...)` to add the `<img>` elements or if we can just use the value of `src` directly.
- [ ] Probably need to update some CSS for those `<img>` elements, too.

Reply:
Cleared the implement the above in this PR. 

Make sure to pull the latest from https://github.com/PAIR-code/lit/commits/dev into your branch before starting on this. 

Need to add a `ImageBytesList` class in two places: first in [types.py](https://github.com/PAIR-code/lit/blob/dev/lit_nlp/api/types.py) then in [lit_types.ts](https://github.com/PAIR-code/lit/blob/dev/lit_nlp/client/lib/lit_types.ts). Both should inherit from `_List`, but can otherwise be empty. Be sure the use the `@registered` decorator with the TypeScript class, this is required so that we can process model and dataset `Spec` information. 

Hopefully the rest goes down smoothly. Let me know if you have any questions.